src/pathfinding_service/model/channel.py

```diff
@@ -34,6 +34,32 @@ def from_raiden(cls, fee_schedule: FeeScheduleRaiden, timestamp: datetime) -> "F
         kwargs.pop("_penalty_func")
         return FeeSchedule(timestamp=timestamp, **kwargs)
 
+    def imbalance_fee_receiver(self, amount: PaymentWithFeeAmount, balance: Balance) -> FeeAmount:
+        if self._penalty_func:
+            # Calculate the mediators balance
+            balance = self._penalty_func.x_list[-1] - balance
+            try:
+                return FeeAmount(
+                    # Mediator is gaining balance on his channel side
+                    round(self._penalty_func(balance + amount) - self._penalty_func(balance))
+                )
+            except ValueError:
+                raise UndefinedMediationFee()
+
+        return FeeAmount(0)
+
+    def imbalance_fee_sender(self, amount: PaymentWithFeeAmount, balance: Balance) -> FeeAmount:
```

Comment:
I have the feeling that `imbalance_fee_receiver` and `backwards_fee_receiver` could be merged in a nice way if `FeeScheduleState.imbalance_fee` was defined slightly differently. But we can keep that for later when we try to reduce duplication between PFS and Raiden.

Reply:
Yeah, there must be a better way. Let's keep that in mind.